api/v1alpha1/managedocs_types.go

```diff
@@ -38,9 +38,24 @@ type ManagedOCSSpec struct {
 	ReconcileStrategy ReconcileStrategy `json:"reconcileStrategy,omitempty"`
 }
 
+type ComponentStatus string
+
+const (
+	ComponentReady   ComponentStatus = "Ready"
+	ComponentPending ComponentStatus = "Pending"
+	ComponentNone    ComponentStatus = "None"
+)
+
+type ManagedOCSComponent struct {
+	Name   string          `json:"name"`
+	Status ComponentStatus `json:"status"`
+}
+
 // ManagedOCSStatus defines the observed state of ManagedOCS
 type ManagedOCSStatus struct {
-	ReconcileStrategy ReconcileStrategy `json:"reconcileStrategy,omitempty"`
+	ReconcileStrategy ReconcileStrategy     `json:"reconcileStrategy,omitempty"`
+	Components        []ManagedOCSComponent `json:"components"`
+	ReconcileCount    uint                  `json:"ReconcileCount,omitempty"`
```

Comment:
Not needed as "generation" is already managed by k8s

Reply:
generation doesn't keep track of the same thing that ReconcileCount is tracking.

Reply:
Ok, so what does it track and why do we need it? 
The CR is a public API (not an implementation detail) and as such every piece of it should have a reasoning

Reply:
"The .metadata.generation value is incremented for all changes, except for changes to .metadata or .status.". I'm tracking all times reconcile was run on this cr.

Reply:
I understand, but this should not be part of the public API.
You have these options:
1. ` generation`
2. `resourceVersion`
3. The last update timestamp